packages/shared/lib/contexts/governance/utils/sortProposals.ts

```diff
@@ -21,16 +22,42 @@ export function sortProposals(proposals: IProposal[], filter: ProposalFilter): I
     return proposals?.sort((proposal1, proposal2) => orderFunction(proposal1, proposal2, isAscending)) ?? []
 }
 
+function sortByPhaseAndMilestone(proposal1: IProposal, proposal2: IProposal, asc: boolean): number {
+    return sortByPhase(proposal1, proposal2, asc) || sortByMilestone(proposal1, proposal2, asc)
+}
+
 function sortByName(proposal1: IProposal, proposal2: IProposal, asc: boolean): number {
     return proposal1.title.toLowerCase() > proposal2.title.toLowerCase() ? (asc ? 1 : -1) : asc ? -1 : 1
 }
 
 function sortByPhase(proposal1: IProposal, proposal2: IProposal, asc: boolean): number {
-    const phaseOrdering = {
-        [ProposalStatus.Upcoming]: 0,
-        [ProposalStatus.Commencing]: 1,
-        [ProposalStatus.Holding]: 2,
-        [ProposalStatus.Ended]: 3,
+    return PROPOSAL_PHASE_ORDERING[proposal1.status] > PROPOSAL_PHASE_ORDERING[proposal2.status]
+        ? asc
+            ? 1
+            : -1
+        : asc
+        ? -1
+        : 1
+}
+
+function sortByMilestone(proposal1: IProposal, proposal2: IProposal, asc: boolean): number {
+    let proposal1Milestone: number
+    let proposal2Milestone: number
+    switch (proposal1.status) {
+        case ProposalStatus.Upcoming:
+            proposal1Milestone = proposal1.milestones.commencing
+            proposal2Milestone = proposal2.milestones.commencing
+            break
+        case ProposalStatus.Commencing:
+            proposal1Milestone = proposal1.milestones.holding
+            proposal2Milestone = proposal2.milestones.holding
+            break
+        case ProposalStatus.Holding:
+        case ProposalStatus.Ended:
+            proposal1Milestone = proposal1.milestones.ended
+            proposal2Milestone = proposal2.milestones.ended
+            break
     }
```

Comment:
```suggestion
            const proposal1Milestone = proposal1.milestones[proposal1.status]
            const proposal2Milestone = proposal2.milestones[proposal2.status]
```

Reply:
We can do this, but it would mean that the milestones are sorted on when the phase started, but IMO it's more useful to sort based off when the current phase ends / next phase starts. If have proposal A that started way long ago and ends in a year and proposal B that started after A but ends in a day, I would think the user wants to see B first?

Reply:
aaaah sorry i didnt see that! i saw that on the last case we could do that, but didnt saw that on the other ones we look at different milestones